middleware/error-handler.js

```diff
@@ -1,10 +1,29 @@
-const { CustomAPIError } = require('../errors')
-const { StatusCodes } = require('http-status-codes')
+const { StatusCodes } = require('http-status-codes');
+
 const errorHandlerMiddleware = (err, req, res, next) => {
-  if (err instanceof CustomAPIError) {
-    return res.status(err.statusCode).json({ msg: err.message })
+  let customError = {
+    statusCode: err.statusCode || StatusCodes.INTERNAL_SERVER_ERROR,
+    msg: err.message || 'Something went wrong, try again later'
+  };
+
+  if (err.name === 'ValidationError') {
+    customError.msg = Object.values(err.errors)
+      .map((item) => item.message)
+      .join(',')
+    customError.statusCode = 400;
+  }
+
+  if (err.code && err.code === 11000) {
+    customError.msg = `Duplicate value entered for ${Object.keys(err.keyValue)} field, please, choose another value`;
+    customError.statusCode = 400;
   }
-  return res.status(StatusCodes.INTERNAL_SERVER_ERROR).json({ err })
-}
 
-module.exports = errorHandlerMiddleware
+  if (err.name === 'CastError') {
```

Comment:
There is another way to encounter CastError besides an id issue.  If the isFavorite value is not boolean.
If you use this body with the update meals
 {  "title":"test",
    "isFavorite": "abc"
}
The returning message doesn't match the actual error
  "msg": "No item is found with id: abc"
  
You can add nested if then's that check the value of err.path and provide the appropriate message

Reply:
Done!